### PR TITLE
More readable code on argument parser

### PR DESCRIPTION
**Pull request by gtozzi**
This should do the same, but with more readability (and arguably in a more "pythonic" way).

### CODE REVIEW

File: Nagstamon/Config.py [hunk 1]
```diff
@@ -179,20 +179,20 @@ def __init__(self):
 
         # Parse the command line
         parser = argparse.ArgumentParser(description='Nagios status monitor for your desktop')
-        parser.add_argument('cfgpath', nargs='*', help='Path for configuration folder')
+        parser.add_argument('cfgpath', nargs='?', help='Path for configuration folder')
```
Comment (HenriWahl) [inline Nagstamon/Config.py]: My last finding was nargs='?' does not work with build script, only nargs='*'

File: Nagstamon/Config.py [hunk 2]
```diff
@@ -179,20 +179,20 @@ def __init__(self):
...
         # if sys.argv is larger than 1
-        if args.cfgpath:
-            # MacOSX related -psn argument by launchd
-            if '-psn' in args.cfgpath:
-                # new configdir approach
-                self.configdir = os.path.expanduser('~') + os.sep + '.nagstamon'
-            else:
-                # allow to give a config file
-                self.configdir = args.cfgpath[0]
+        if args.psn:
```
Comment (HenriWahl) [inline Nagstamon/Config.py]: what if there is no arg '-psn'? I assume this check would fail on anything not OSX

File: Nagstamon/Config.py (hunk 1 above)
Comment (gtozzi) [inline Nagstamon/Config.py]: You used nargs='*' and then config[0] as a workaround for not adding a '-psn' argument. This is supposed to work the same. The build-script related part should be only the known_args() call.

File: Nagstamon/Config.py (hunk 2 above)
Comment (gtozzi) [inline Nagstamon/Config.py]: Yes, if there is no -psn this is False. Isn't how it is supposed to work?

Comment (HenriWahl) [inline Nagstamon/Config.py]: No it will crash if args does not have the property .psn - this is why I checked if it has it.

Comment (gtozzi) [inline Nagstamon/Config.py]: It does not happen to me. Can you please paste the trace?

Comment (HenriWahl) [inline Nagstamon/Config.py]: At least I had it yesterday with another argument. I will check this - right now I try to fix the build script for Debian which I already finished.

Comment (gtozzi) [inline Nagstamon/Config.py]: I am sorry I've got confused. I wrote this code this morning, how could it crash yesterday?
Anyway, take your time ;)

Thank you for your work.

Comment (HenriWahl) [inline Nagstamon/Config.py]: I tried yesterday this args.something where it crashed if there was no .something given as --something at the commandline.

Comment (HenriWahl) [inline Nagstamon/Config.py]: But I should not do three things at the same time... :-)

Comment (gtozzi) [inline Nagstamon/Config.py]: Please just give it a try when you'll have time ;)

Comment (HenriWahl) [inline Nagstamon/Config.py]: Promised!

**Comment by gtozzi** [comment]
I've added #189 to this pull request ;)

**Comment by HenriWahl** [comment]
Works great, thanks! I will just look forward to add GUI option and consider https://github.com/HenriWahl/Nagstamon/issues/189#issuecomment-191244586.

**Comment by gtozzi** [comment]
Thank to you ;)
About #189 comment, I have already included the username in the files, so it should also address that.
Probably the GUI options is really not needed anymore. Unless somebody really wants to run two instances on the same machine from the same user on the same config :P

**Comment by HenriWahl** [comment]
But different users will overwrite the same file, right?

**Comment by gtozzi** [comment]
Yes, if multiple users are using the same config folder at the same time, the lock file gets overwritten.
It's not the same for the config itself?

**Comment by HenriWahl** [comment]
Errm.. yes. :-)

**Pull request merged**